### PR TITLE
Configure envoy to use statsd to report metrics

### PR DESCRIPTION
**Pull request by kramerul**

### CODE REVIEW

**Comment by ndhanushkodi** [comment]
hey @kramerul, can you let us know some more context about your use case for this change?

**Comment by loewenstein** [comment]
Hi @ndhanushkodi, we co-deploy a [telegraf](https://github.com/influxdata/telegraf) bosh release to get a [local statsd sink](https://github.com/influxdata/telegraf/blob/master/plugins/inputs/statsd/README.md) that [pushes the data](https://github.com/influxdata/telegraf/blob/master/plugins/inputs/influxdb/README.md) to our monitoring stack.
We will potentially add [datadog statsd sink](https://www.envoyproxy.io/docs/envoy/latest/api-v2/config/metrics/v2/stats.proto#config-metrics-v2-dogstatsdsink) as alternative in a later PR.

**Comment by loewenstein** [comment]
Hi Shannon,

We will potentially have standalone deployments in addition to those side-by-side with CF, so we looked for a generic solution. But you have a valid point regarding scalability. We will definitely look into loggregator for the deloyment running beside CF. 

This doesn't change the need for this PR though, correct?

Both Envoy and Telegraf apparently support an extension to the statsd protocol. The [datadog statsd sink](https://www.envoyproxy.io/docs/envoy/latest/api-v2/config/metrics/v2/stats.proto#config-metrics-v2-dogstatsdsink) will deliver statsd metrics with tags and the [telegraf statsd plugin](https://github.com/influxdata/telegraf/tree/master/plugins/inputs/statsd) has a switch to understand that statsd extension.
Do you know, if the [statsd-injector-release](https://github.com/cloudfoundry/statsd-injector-release) has support for this extension as well?

Thanks and best regards
Jan

File: jobs/envoy/spec [hunk 1]
```diff
@@ -16,3 +16,17 @@ consumes:
 - name: frontend_tls_keypairs
   type: frontend_tls_keypairs
   optional: true
+
+properties:
+  statsd.enabled:
+    description: Should a statsd sink be configured
+    default: false
+  statsd.address:
```
Comment (shalako) [inline jobs/envoy/spec:27]: Does the statsd client support a DNS name?

Comment (shalako) [reply jobs/envoy/spec:27]: Oh, I see the default in localhost. So the sink will be colocated, like a BOSH add-on, and be configured independently with the address of the remote aggregator?

**Comment by loewenstein** [comment]
@shalako yes, the idea is to work with a local agent. Telegraf could act as such, but the statsd-injector from loggregator could as well.

The only thing is, we noticed plain statsd could be suboptimal for overall metrics (in contrast to envoy cluster specific ones). Envoy has support for tags with the datadog statsd sink, Telegraf has as well, but I couldn't find anything in that regard for the statsd-injector.

**Comment by shalako** [comment]
Thank you for your clarification. Support for emitting via statsd to an arbitrary sink makes sense. 

I would also want per-cluster stats. @jtuchscherer is the tagging that @loewenstein mentioned on your radar for statsd-injector?

**Comment by jtuchscherer** [comment]
@loewenstein @shalako no, the tagging in the statsd-injector has not been on my radar yet, but it is now. 
I created this issue [here](https://github.com/cloudfoundry/statsd-injector/issues/5), please feel free to comment.

File: jobs/envoy/spec (hunk 1 above)
Comment (kramerul) [reply jobs/envoy/spec:27]: Exactly. We are using a colocated statsd.

**Review by XanderStrike — APPROVED** [review]

**Comment by jtuchscherer** [comment]
Datadog style tagging is now supported in the latest statsd-injector-release:
https://github.com/cloudfoundry/statsd-injector-release/releases/tag/v1.8.0

**Pull request merged**